File: frontend/tips/6-open-new-window/code/csharp/NewWindowNewTabTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.Firefox;

namespace csharp
{
    [TestClass]
    public class NewWindowNewTabTest
    {
        IWebDriver _driver;

        [TestInitialize]
        public void SetUp()
        {
            _driver = new FirefoxDriver();
        }

        [TestCleanup]
        public void TearDown()
        {
            _driver.Quit();
        }

        [TestMethod]
        public void NewWindowTest()
        {
            _driver.Navigate().GoToUrl("https://the-internet.herokuapp.com/windows");
            _driver.SwitchTo().NewWindow(WindowType.Window);
            _driver.Navigate().GoToUrl("https://the-internet.herokuapp.com/typos");
            Assert.AreEqual(2, _driver.WindowHandles.Count);
        }

        [TestMethod]
        public void NewTabTest()
        {
            _driver.Navigate().GoToUrl("https://the-internet.herokuapp.com/windows");
            _driver.SwitchTo().NewWindow(WindowType.Tab);
            _driver.Navigate().GoToUrl("https://the-internet.herokuapp.com/typos");
            Assert.AreEqual(2, _driver.WindowHandles.Count);
        }
    }
}

File: frontend/tips/6-open-new-window/code/java/src/test/java/com/elemental/selenium/NewWindowNewTabTest.java
package com.elemental.selenium;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.Assertions;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WindowType;
import org.openqa.selenium.firefox.FirefoxDriver;

public class NewWindowNewTabTest {
    WebDriver driver;

    @BeforeEach
    public void beforeEach() {
        driver = new FirefoxDriver();
    }

    @AfterEach
    public void afterEach() {
        if (driver != null) {
          driver.quit();
        }
    }

    @Test
    public void newWindow() throws InterruptedException {
        driver.get("https://the-internet.herokuapp.com/windows");
        driver.switchTo().newWindow(WindowType.WINDOW);
        driver.get("https://the-internet.herokuapp.com/typos");
        Assertions.assertEquals(2, driver.getWindowHandles().toArray().length);
    }

    @Test
    public void newTab() throws InterruptedException {
        driver.get("https://the-internet.herokuapp.com/windows");
        driver.switchTo().newWindow(WindowType.TAB);
        driver.get("https://the-internet.herokuapp.com/typos");
        Assertions.assertEquals(2, driver.getWindowHandles().toArray().length);
    }

}